cloudfront/main.tf

```diff
@@ -21,8 +21,9 @@ resource "aws_cloudfront_distribution" "distribution" {
   viewer_certificate {
     cloudfront_default_certificate = var.certificate_arn == null
 
-    acm_certificate_arn = var.certificate_arn
-    ssl_support_method  = var.certificate_arn != null ? "sni-only" : null
+    acm_certificate_arn      = var.certificate_arn
+    ssl_support_method       = var.certificate_arn != null ? "sni-only" : null
+    minimum_protocol_version = var.certificate_arn == null ? null : var.ssl_policy
```

Comment:
NIT: Consider switching the clauses of the ternary operator, it would be consistent with the previous statement.

Reply:
swapped in d8e4ead